hibernate-core/src/test/java/org/hibernate/test/hql/HQLTest.java

```diff
@@ -1538,4 +1501,11 @@ public void testComponentNoAlias() throws Exception {
 		compileWithAstQueryTranslator( "from Human where name.first = 'Gavin'", false);
 	}
 
+	@Test
+	public void testCountDistinctOfSubquery(){
+		createNewQueryTranslator(
+				"select count (DISTINCT (SELECT function('lower', 'DAY') from Bar)) from Bar"
```

Comment:
You are right, even SQL-99 disallows this: https://mariadb.com/kb/en/sql-99/set-functions/
Still, it's silly because you could workaround that by introducing a table alias/derived table for the subquery and use that instead which is probably why PostgreSQL directly supports that syntax.

Oracle seems to support it: https://docs.oracle.com/cd/B28359_01/server.111/b28286/functions032.htm#SQLRF00624
SQLServer does not like it: http://www.sqlteam.com/article/aggregating-correlated-sub-queries
DB2 again seems to like it: http://www.ibm.com/support/knowledgecenter/SSEPEK_11.0.0/sqlref/src/tpc/db2z_aggregatefunctionsintro.html
PostgreSQL also allows it: ```WITH a AS (select 1) SELECT COUNT((SELECT 1 FROM a)) FROM a```
Not sure about MySQL